File: Backend/Travle.Model/Responses/DestinationImageResponse.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// Metadata for one of a destination's images — never the bytes. The full image is fetched from the
    /// dedicated endpoint (<c>GET /Destinations/{destinationId}/images/{id}</c>) so list/detail payloads
    /// stay light (§8.2). <see cref="SortOrder"/> gives the gallery/edit-grid its order.
    /// </summary>
    public class DestinationImageResponse
    {
        public int Id { get; set; }
        public string ContentType { get; set; } = string.Empty;
        public int SortOrder { get; set; }
    }
}

File: UI/travle_core/lib/src/models/destination_image_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'destination_image_response.g.dart';

/// Metadata for one of a destination's images (mirrors the backend
/// `DestinationImageResponse`) — never the bytes. Fetch the full image from
/// `GET /Destinations/{destinationId}/images/{id}`.
@JsonSerializable()
class DestinationImageResponse {
  DestinationImageResponse({
    required this.id,
    required this.contentType,
    required this.sortOrder,
  });

  final int id;
  final String contentType;
  final int sortOrder;

  factory DestinationImageResponse.fromJson(Map<String, dynamic> json) =>
      _$DestinationImageResponseFromJson(json);

  Map<String, dynamic> toJson() => _$DestinationImageResponseToJson(this);
}
